web/src/app/components/share-dialog/share-dialog.component.ts

```diff
@@ -103,6 +97,8 @@ export class ShareDialogComponent {
 
     // Clear "Add data collector" field.
     this.addUserForm.setValue({email: '', role: Role.DATA_COLLECTOR});
+
+    this.onSaveClicked();
```

Comment:
Could you please also add unit tests for the various components? You can see examples of tests we wrote for other components by searching for `spec.ts`. @DaoyuT may be able to point you to a good example.